tests/components/prometheus/test_init.py

```diff
@@ -87,6 +87,71 @@
 PROMETHEUS_PATH = "homeassistant.components.prometheus"
 
 
+@dataclass(frozen=True)
+class EntityTestMetricInfo:
+    """Class for all entity data in a single prometheus metric."""
+
+    metric_name: str
+    domain: str
+    friendly_name: str
+    object_id: str
+    metric_value: Any | None = None
+    mode: str | None = None
+    state: str | None = None
+    action: str | None = None
+
+    @classmethod
+    def from_entity_id(cls, entity_id: str, **kwargs) -> Self:
+        """Construct EntityTestMetricInfo using full entity_id instead of parts."""
+        domain, object_id = entity_id.split(".")
+        return cls(
+            metric_name=kwargs["metric_name"],
+            domain=domain,
+            friendly_name=kwargs["friendly_name"],
+            object_id=object_id,
+            metric_value=kwargs.get("metric_value"),
+            mode=kwargs.get("mode"),
+            state=kwargs.get("state"),
+            action=kwargs.get("action"),
+        )
+
+    @property
+    def entity(self) -> str:
+        """Generate entity_id from components."""
+        return f"{self.domain}.{self.object_id}"
+
+    def get_full_metric_string(self) -> str:
+        """Convert metric info into a valid prometheus text string."""
+        final_metric_value = (
+            f" {self.metric_value}" if self.metric_value is not None else ""
+        )
+        final_action_value = (
+            f'action="{self.action}",' if self.action is not None else ""
+        )
+        final_mode_value = f',mode="{self.mode}"' if self.mode is not None else ""
+        final_state_value = f',state="{self.state}"' if self.state is not None else ""
+        return (
+            f"{self.metric_name}{{"
+            f"{final_action_value}"
+            f'domain="{self.domain}",'
+            f'entity="{self.entity}",'
+            f'friendly_name="{self.friendly_name}"'
+            f"{final_mode_value}"
+            f"{final_state_value}"
+            f"}}{final_metric_value}"
+        )
+
+
+def _assert_metric_present(body, metric_info: EntityTestMetricInfo) -> None:
+    """Assert metrics text body contains metric for entity."""
+    assert metric_info.get_full_metric_string() in body
+
+
+def _assert_metric_not_present(body, metric_info: EntityTestMetricInfo) -> None:
+    """Assert metrics text body does not contain metric for entity."""
+    assert metric_info.get_full_metric_string() not in body
```

Comment:
I wonder moving these to be methods on the Metric class would make tests more readable?

Reply:
Great idea! Done!